olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -97,6 +97,22 @@ client::CancellationToken VersionedLayerClientImpl::GetPartitions(
                    std::move(partitions_task), std::move(callback));
   };
 
+  if (request.GetFetchOption() == CacheWithUpdate) {
+    request.WithFetchOption(FetchOptions::OnlineIfNotFound);
```

Comment:
Why do you need to change FetchOptions to OnlineIfNotFound?

Reply:
ScheduleFetch function in case of   if (request.GetFetchOption() == FetchOptions::CacheWithUpdate)  will call schedule_task two times. For returning error is is not needed.

`  if (request.GetFetchOption() == FetchOptions::CacheWithUpdate) {
    auto cache_token =
        schedule_task(request.WithFetchOption(FetchOptions::CacheOnly),
                      std::forward<Callback>(callback));
    auto online_token = schedule_task(
        std::move(request.WithFetchOption(FetchOptions::OnlineOnly)), nullptr);
`

Reply:
But then we maybe should not use `ScheduleFetch()` if we need to trick it so that it behaves correctly. Just use AddTask.